test/error-messages/nested-invalid-value.txt

```diff
@@ -0,0 +1,9 @@
+Invalid value
+
+concat :: Array a -> Array a -> Array a
+                     ^^^^^^^
+                        1
+
+1) "ab" :: String
+
+The value at position 1 is not a member of (Array Number).
```

Comment:
This is in contrast to `nested-type-var-conflict`
```
Type-variable conflict

concat :: Array a -> Array a -> Array a
                ^          ^
                1          2

1) 1 :: Number

2) "a" :: String

Since there is no type of which both values are members, the type-variable constraint has been violated.
```
Since it doesn't matter what type `a` is `nested-invalid-value` as "ab" will always be in violation, I wonder if it would be better to use `???` as in:
```
Invalid value

concat :: Array a -> Array a -> Array a
                     ^^^^^^^
                        1

1) "ab" :: String

The value at position 1 is not a member of (Array ???).
```
Currently we don't pass multiple positions to `invalidValue`, so it would be a bit of work to show both. I haven't considered `???` instead of the actual value, not sure off the top of my head how difficult it would be to determine when that would be appropriate.